src/server/util/GetPostRouterHandler.ts

```diff
@@ -0,0 +1,33 @@
+import { MethodNotAllowedHttpError } from '../../util/errors/MethodNotAllowedHttpError';
+import type { HttpHandler, HttpHandlerInput } from '../HttpHandler';
+import { RouterHandler } from './RouterHandler';
+
+/**
+ * A router handler that handles both GET and POST for
+ * a single route. This could be used to render a form
+ * and handle its submission.
+ */
+export class GetPostRouterHandler extends RouterHandler {
+  private readonly getHandler: HttpHandler;
+  private readonly postHandler: HttpHandler;
```

Comment:
Shall we just make this an abstract class, where `handleGet` and `handlePost` are methods?

Reply:
Ah, but `RenderEjsHandler` below might prove me wrong then. Or maybe not; this can just be a helper function then.